frontend/src/components/AuthCallbackHandler.vue

```diff
@@ -1,13 +1,19 @@
 <script setup lang="ts">
 import router from '@/router';
 import authService from '@/services/AuthService';
+import { getNavigationPath } from '@/store/NavigationState';
 
 /**
  * This component is to deal with Auth callback.
  */
 
 authService.methods.handlePostLogin().then((data) => {
-    router.push('/dashboard');
+    const navigationPath = getNavigationPath();
+    if (navigationPath) {
+        router.push(navigationPath);
+    } else {
+        router.push('/dashboard');
+    }
```

Comment:
Just wondering if we `setNavigationPath(to.path)` in the `beforeEachRouteHandler`, then maybe we'll always have the navigationPath? no need to check if else here?